internal/models/challenge.go

```diff
@@ -33,12 +33,15 @@ func NewChallenge(factor *Factor, ipAddress string) *Challenge {
 	return challenge
 }
 
-func FindChallengeByChallengeID(tx *storage.Connection, challengeID uuid.UUID) (*Challenge, error) {
-	challenge, err := findChallenge(tx, "id = ?", challengeID)
-	if err != nil {
+func FindChallengeByID(tx *storage.Connection, challengeID uuid.UUID) (*Challenge, error) {
```

Comment:
nit: maybe name the first arg `conn` instead of `tx` since tx is short for transaction?
```suggestion
func FindChallengeByID(conn *storage.Connection, challengeID uuid.UUID) (*Challenge, error) {
```